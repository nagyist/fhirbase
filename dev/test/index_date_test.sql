--db:fhirb -e

SET escape_string_warning=off;
--{{{

SELECT index_as_date('{}', '{birthDate}','dateTime');
SELECT index_as_date('{"birthDate": "1980"}', '{birthDate}','dateTime');
SELECT index_as_date('{"birthDate": "1980-03"}', '{birthDate}','dateTime');
SELECT index_as_date('{"birthDate": "1980-03-05"}', '{birthDate}','dateTime');
SELECT index_as_date('{"birthDate": "1980-03-05 12"}', '{birthDate}','dateTime');
SELECT index_as_date('{"birthDate": "1980-03-05 12:30"}', '{birthDate}','dateTime');


SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30"}}', '{issued}', 'Period');
SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30", "end":"2014-12-05"}}', '{issued}', 'Period');
SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30", "end":"2014-12-05"}}', '{issued}', 'Period');
SELECT index_as_date('{"issued": [{"start": "2014-01-05 12:30", "end":"2014-02-05"},{"start":"2014-03", "end":"2014-07"}]}', '{issued}', 'Period');

SELECT index_as_date('{"issued": [{"start": "2014-01-05 12:30", "end":"2014-02-05"},{"start":"2014-03", "end":"2014-07"}]}', '{issued}', 'Period');


\set careplan `cat test/fixtures/careplan.json`
SELECT index_as_date(:'careplan', '{activity,simple,timingSchedule}','Schedule');
--}}}

