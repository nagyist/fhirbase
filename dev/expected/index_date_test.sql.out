SET escape_string_warning=off;
SET
SELECT index_as_date('{}', '{birthDate}','dateTime');
 index_as_date 
---------------
 
(1 row)

SELECT index_as_date('{"birthDate": "1980"}', '{birthDate}','dateTime');
                    index_as_date                    
-----------------------------------------------------
 ["1980-01-01 00:00:00+03","1980-12-31 23:59:59+03"]
(1 row)

SELECT index_as_date('{"birthDate": "1980-03"}', '{birthDate}','dateTime');
                    index_as_date                    
-----------------------------------------------------
 ["1980-03-01 00:00:00+03","1980-03-31 23:59:59+03"]
(1 row)

SELECT index_as_date('{"birthDate": "1980-03-05"}', '{birthDate}','dateTime');
                    index_as_date                    
-----------------------------------------------------
 ["1980-03-05 00:00:00+03","1980-03-05 23:59:59+03"]
(1 row)

SELECT index_as_date('{"birthDate": "1980-03-05 12"}', '{birthDate}','dateTime');
                    index_as_date                    
-----------------------------------------------------
 ["1980-03-05 12:00:00+03","1980-03-05 12:59:59+03"]
(1 row)

SELECT index_as_date('{"birthDate": "1980-03-05 12:30"}', '{birthDate}','dateTime');
                    index_as_date                    
-----------------------------------------------------
 ["1980-03-05 12:30:00+03","1980-03-05 12:30:59+03"]
(1 row)

SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30"}}', '{issued}', 'Period');
        index_as_date        
-----------------------------
 ["2014-01-05 12:30:00+04",)
(1 row)

SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30", "end":"2014-12-05"}}', '{issued}', 'Period');
                    index_as_date                    
-----------------------------------------------------
 ["2014-01-05 12:30:00+04","2014-12-05 23:59:59+04"]
(1 row)

SELECT index_as_date('{"issued": {"start": "2014-01-05 12:30", "end":"2014-12-05"}}', '{issued}', 'Period');
                    index_as_date                    
-----------------------------------------------------
 ["2014-01-05 12:30:00+04","2014-12-05 23:59:59+04"]
(1 row)

SELECT index_as_date('{"issued": [{"start": "2014-01-05 12:30", "end":"2014-02-05"},{"start":"2014-03", "end":"2014-07"}]}', '{issued}', 'Period');
                    index_as_date                    
-----------------------------------------------------
 ["2014-01-05 12:30:00+04","2014-07-31 23:59:59+04"]
(1 row)

SELECT index_as_date('{"issued": [{"start": "2014-01-05 12:30", "end":"2014-02-05"},{"start":"2014-03", "end":"2014-07"}]}', '{issued}', 'Period');
                    index_as_date                    
-----------------------------------------------------
 ["2014-01-05 12:30:00+04","2014-07-31 23:59:59+04"]
(1 row)

SELECT index_as_date( E'{"resourceType":"CarePlan","text":{"status":"additional","div":"<div>\\n           <p> A simple care plan to indicate a patient taking their weight once a day because of obesity.\\n            Some Notes: </p>\\n            <ul>\\n            <li>It would be good to have some way of specifying/coding a goal. e.g. what the target weight is</li>\\n            <li>In the codeable concepts I''ve been lazy and just put the text. There should, of course, be a code behind these</li>\\n        </ul>\\n        </div>"},"contained":[{"resourceType":"Condition","id":"p1","subject":{"reference":"Patient/example","display":"Peter James Chalmers"},"code":{"text":"Obesity"},"status":"confirmed"},{"resourceType":"Practitioner","id":"pr1","name":{"family":["Dietician"],"given":["Dorothy"]},"specialty":[{"text":"Dietician"}]}],"patient":{"reference":"Patient/example","display":"Peter James Chalmers"},"status":"active","period":{"end":"2013-01-01"},"concern":[{"reference":"#p1","display":"obesity"}],"participant":[{"role":{"text":"responsiblePerson"},"member":{"reference":"Patient/example","display":"Peter James Chalmers"}},{"role":{"text":"adviser"},"member":{"reference":"#pr1","display":"Dorothy Dietition"}}],"goal":[{"description":"Target weight is 80 kg. Note: be nice if this could be coded"}],"activity":[{"prohibited":false,"simple":{"category":"observation","code":{"text":"a code for weight measurement"},
  "timingSchedule":{"event":[{"start": "2014-03-05"}]},"performer":[{"reference":"Patient/example","display":"Peter James Chalmers"}]}}]}', '{activity,simple,timingSchedule}','Schedule');
        index_as_date        
-----------------------------
 ["2014-03-05 00:00:00+04",)
(1 row)

